### DISCUSSION TITLE
Commercially available batteries with compatible BMSs

### DISCUSSION THREAD

**Discussion by HarrisMarine** [Louisvdw/dbus-serialbattery#627] [Q&A]
Hi All,

As per the thread title. What prebuilt commercial batteries have a dbus serial battery compatible BMS? 

My reason for asking is the new ISO/ABYC rules for lithium batteries on boats. Which states the batteries must be able to control the charging and discharging devices as well as notify the boat operator of any pre-shutdown condition. I’m predicting this will result in a couple of disgruntled boat owners. So the dbus serial battery driver could be a solution.

I’m wondering which batteries can be opened up and connected via this method?

Thanks in advance.

Keith

**Comment by dfch** [comment]
Ho @HarrisMarine 

 I am not sure such a list exists or at least I cannot find one. However, I do not think the developers of the dbus-serialbattery driver aim to support specific batteries but only BMS types. 

As an example, [Smart BMS](https://louisvdw.github.io/dbus-serialbattery/general/supported-bms/#-smart-bms-) is supported, and therefore batteries from Liontron could also work (as they are using that BMS). 

So, you might want to check [Supported BMS](https://louisvdw.github.io/dbus-serialbattery/general/supported-bms/) and have a look at different commercial available batteries.

**Reply by HarrisMarine** [reply]
Hi dfch,

Thanks for your reply. 

I have since found a couple of battery manufacturers that are using JBD BMSs, they both are saying they intend to release Victron compatible Canbus batteries in the very near future. I’m not sure if that’s utilizing d-bus or not.

**Reply by dfch** [reply]
Every battery in Venus OS needs to publish its data via dbus. Otherwise I do not think it will not be recognised as a battery. You would then not need dbus-serialbattery, but a different driver, such as [VeCanSetup](https://github.com/kwindrem/VeCanSetup) (and a CAN adapter), if you are running on Raspberry PI (devices like the Cerbo GX have built-in dedicated VE.Can ports).

However, BYD LVS Battery-Box Premium (just an example) is officially supported / compatible with Victron and also communicates vai CAN to Venus OS (and publishes its data on dbus). But with this implementation I do not find any paths in dbus exposed that could allow me to disable discharging. But of course, you would at least get the alarms.

So, it really depends on the implemenation of the driver and not only on the communication protocol (regardless of RS485 or CAN).

Hope this helps


ps JBD is already supported via dbus-serialbattery as it seems. So, you could start testing with these batteries right away.
